### PR TITLE
Add script to help with vault interactions

### PR DESCRIPTION
**Pull request by yufengwng**
No more overwriting vault credentials 😢 . Introducing a new script that will make it harder to directly write into vault, forcing the user to go through a few confirmations. Plus some nifty `read` commands.

You can symlink this to your `$PATH` and then invoke it in place of your typical `vault` command, e.g.
```
$ vaultcli read <path> -r .dbUrl
$ vaultcli write <path> data.json
$ vaultcli write <path> --value cert.key
```

### CODE REVIEW

**Review by zyme — COMMENTED** [review]
Awesome!  Handles cert paths as well :)

**Review by zyme — COMMENTED** [review]
Let's make a decision at process and standards to do something dramatic like _not_ use lower level `vault [read | write]`...

File: scripts/vaultcli [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  $NAME write <PATH> <FILE> [OPTIONS]...
+
+Writes given file as data to vault path after passing confirmation. One
+confirmation step is opening the given data file in the 'less' pager so user
+can visually verify the data to be written. Before writing, a backup of
+existing data in vault will be written to a file in the current directory.
+
+FLAGS:
+  -h, --help    Print this additional help
+      --value   Write whole contents of data file into the 'value' key
```
Comment (yufengwng) [inline scripts/vaultcli:64]: I realized that we have some paths in vault that stores entire file content as a blob under the `value` key in that path. We can read these using existing `vaultcli read .. -r .value` functionality, but writes have a different syntax (see L200), so I added a flag to help with writing values.

**Review by zyme — COMMENTED** [review]
/pullapprove-test

```yml
version: 2

notifications:
  - when: pull_request.opened
    if: "base.ref == 'test' or base.ref == 'staging' or base.ref == 'production'"
    comment: |
      Hey @{{ author }}, looks like we're doing a deploy to @{{base}}.  Cool!
      The reviewers will let you know when it's approved.
  - when: pullapprove.approved
    if: "base.ref == 'test' or base.ref == 'staging' or base.ref == 'production'"
    comment: |
      Thanks @{{ author }}, your @{{base}} deployment has been approved.  Please follow the merge instructions.

groups:
  test-deployment:
    required: 2
    users:
      - zyme
      - yufengwng

    conditions:
    - "base.ref == 'test'"
```

**Review by zyme — COMMENTED** [review]
/pullapprove-test

```
version: 3

notifications:
  - when: pull_request.opened
    if: "base.ref == 'test' or base.ref == 'staging' or base.ref == 'production'"
    comment: |
      Hey @{{ author }}, looks like we're doing a deploy to @{{base}}.  Cool!
      The reviewers will let you know when it's approved.
  - when: pullapprove.approved
    if: "base.ref == 'test' or base.ref == 'staging' or base.ref == 'production'"
    comment: |
      Thanks @{{ author }}, your @{{base}} deployment has been approved.  Please follow the merge instructions.

groups:
  test-deployment:
    required: 2
    users:
      - zyme
      - yufengwng

    conditions:
    - "base.ref == 'test'"
```

**Review by zyme — COMMENTED** [review]
/pullapprove-test

```yml

version: 3

notifications:
  - when: pull_request.opened
    if: "base.ref == 'test' or base.ref == 'staging' or base.ref == 'production'"
    comment: |
      Hey @{{ author }}, looks like we're doing a deploy to @{{base}}.  Cool!
      The reviewers will let you know when it's approved.
  - when: pullapprove.approved
    if: "base.ref == 'test' or base.ref == 'staging' or base.ref == 'production'"
    comment: |
      Thanks @{{ author }}, your @{{base}} deployment has been approved.  Please follow the merge instructions.

groups:
  test-deployment:
    conditions:
    - "base.ref == 'test'"
    reviews:
      required: 1
    reviewers:
      users:
      - zyme
      - yufengwng

```

**Review by zyme — APPROVED** [review]

**Review by zyme — APPROVED** [review]

**Comment by yufengwng** [comment]
please run tests

**Pull request merged**